persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/CassandraPersistenceActivator.java

```diff
@@ -121,6 +124,10 @@ private Config makeConfig() throws IOException {
 
   @Override
   public void start(BundleContext context) {
+    ServiceReference<?> metricsReference = context.getServiceReference(Metrics.class.getName());
```

Comment:
Doesn't this need a service listener?

Reply:
I had a comment asking about that, but it got lost in the formatting changes: do we need a service listener for every bundle, or only those that might be changed at runtime?

Reply:
The problem is initialization ordering can cause other modules to not be able during the initialization of this module. Also, it'll enable hot reloading.